### PR TITLE
Save modal/dialog

### LINKED ISSUE #114: Add a modal for branch name and commit message forms
Modal opens when user clicks save buttons. If user saves to current branch, do not display branch name form.

### PR DESCRIPTION
**Pull request by jakekall**
Closes #114 

- Created a save dialog that allows the user to select between current and new branch and also write a commit message
- Changed the save button to use Material-UI

Currently the branch name and commit message are only logged to console because backend does not support this feature yet. This can be changed after #113 and #115 are merged.

Known issue:
When the save dialog is opened, warning "findDOMNode is deprecated in StrictMode" will be output to console. This is a problem within the library and as far as I understand only way to get rid of it is to either not use strict mode at all or to update Material-UI to alpha version. The functionality of the component is not affacted in any way and the warning should not show up in production (since strict mode won't be enabled) so my suggestion is to just ignore it.

### CODE REVIEW

**Comment by jakekall** [comment]
Made a separate issue about integrating this with the upcoming backend changes -> #121

**Review by guotin — APPROVED** [review]
Nice smooth animations

File: frontend/src/components/SaveDialog.tsx [hunk 1]
```diff
@@ -0,0 +1,124 @@
...
+        <Grid container direction="column">
+          <Grid item>
+            <FormControl component="fieldset">
+              <FormLabel component="legend">Select branch</FormLabel>
+              <RadioGroup
+                aria-label="branch selection"
+                value={createNewBranch}
+                onChange={(event) =>
+                  setCreateNewBranch(
+                    (event.target as HTMLInputElement).value === 'true'
```
Comment (tulma95) [inline frontend/src/components/SaveDialog.tsx:60]: I think the event gets right type automatically without so ` as HTMLInputElement` is redundant.

File: frontend/src/components/SaveDialog.tsx [hunk 2]
```diff
@@ -0,0 +1,124 @@
...
+          <Grid item>
+            <Collapse in={createNewBranch}>
+              <TextField
+                label="Branch name"
+                variant="outlined"
+                margin="normal"
+                fullWidth
+                value={branchName}
+                onChange={(event) =>
+                  setBranchName((event.target as HTMLInputElement).value)
```
Comment (tulma95) [inline frontend/src/components/SaveDialog.tsx:86]: Same as above

File: frontend/src/components/SaveDialog.tsx [hunk 3]
```diff
@@ -0,0 +1,124 @@
...
+        <TextField
+          margin="normal"
+          label="Commit message"
+          multiline
+          rows={4}
+          variant="outlined"
+          fullWidth
+          value={commitMessage}
+          onChange={(event) =>
+            setCommitMessage((event.target as HTMLInputElement).value)
```
Comment (tulma95) [inline frontend/src/components/SaveDialog.tsx:102]: Same as above

**Review by tulma95 — APPROVED** [review]
Found few little things with types check comments. Otherwise looks good

**Pull request merged**